services/ubuntu/ubuntu.tester.js

```diff
@@ -1,15 +1,17 @@
 'use strict'
 
-const {
-  isVPlusDottedVersionNClausesWithOptionalSuffixAndEpoch,
-} = require('../test-validators')
+const Joi = require('@hapi/joi')
 const t = (module.exports = require('../tester').createServiceTester())
 
+const isUbuntuVersion = Joi.string().regex(
+  /^v(\d+:)?\d+(\.\d+)*([\w\\.]*)?([-+~].*)?$/
+)
```

Comment:
I elected to not add this to `test-validators` as it seems _very_ specific to ubuntu AFAICT. The ubuntu service class is using the much more relaxed `Joi.string().required()`, so I considered using this in the service class schema as well, but decided not to (at least for now) as I wasn't sure the full breadth of versions that are allowed and I didn't want to break any existing badges. 

I could easily be convinced to update the service class schema though as well if anyone feels strongly we should 😄

Reply:
> I wasn't sure the full breadth of versions that are allowed and I didn't want to break any existing badges.
Sounds very reasonable for me.